bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -261,63 +263,30 @@ public synchronized void login() {
+                loginOAUTH();
             }
 
-            ContentResponse response = sendRequestBuilder("login", HttpMethod.POST)
-                    .content(new StringContentProvider(urlParameters),
-                            "application/x-www-form-urlencoded; charset=UTF-8")
-                    .send();
-
-            if (logger.isTraceEnabled()) {
-                logger.trace("Login response: {}", response.getContentAsString());
+            if (thingConfig.isDevMode()) {
+                initializeLocalMode();
             }
 
-            SomfyTahomaLoginResponse data = gson.fromJson(response.getContentAsString(),
-                    SomfyTahomaLoginResponse.class);
-
-            lastLoginTimestamp = Instant.now();
-
-            if (data == null) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        "Received invalid data (login)");
-            } else if (!data.getErrorCode().isEmpty()) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, data.getError());
-                if (data.getError().startsWith(TOO_MANY_REQUESTS)) {
-                    setTooManyRequests();
-                }
+            String id = registerEvents();
+            if (id != null && !UNAUTHORIZED.equals(id)) {
+                eventsId = id;
+                logger.debug("Events id: {}", eventsId);
+                updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                        isDevModeReady() ? "LAN mode" : cloudFallback ? "Cloud mode fallback" : "Cloud mode");
             } else {
-                if (thingConfig.isDevMode()) {
-                    initializeLocalMode();
-                }
-
-                String id = registerEvents();
-                if (id != null && !UNAUTHORIZED.equals(id)) {
-                    eventsId = id;
-                    logger.debug("Events id: {}", eventsId);
-                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
-                            isDevModeReady() ? "LAN mode" : cloudFallback ? "Cloud mode fallback" : "Cloud mode");
-                } else {
-                    logger.debug("Events id error: {}", id);
-                    if (!thingConfig.isDevMode()) {
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                                "unable to register events");
-                    } else {
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                                "LAN mode is not properly configured");
-                        logger.debug("Forcing the gateway discovery");
-                        discoverGateway();
-                    }
-                }
+                logger.debug("Events id error: {}", id);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "unable to register events");
             }
+
```

Comment:
Ultra minor: please remove this blank line

Reply:
you mean the blank line at the end of the file? mvn spotless:apply will always add this line to the end...

Reply:
No, just the line I pointed, line 289 before your last change.

Reply:
fixed